crates/bevy_render/src/mesh/mesh/conversions.rs

```diff
@@ -45,325 +46,120 @@ impl FromVertexAttributeError {
-    fn from(vec: Vec<[f32; 2]>) -> Self {
-        VertexAttributeValues::Float32x2(vec)
-    }
-}
-
-impl From<Vec<[i32; 2]>> for VertexAttributeValues {
-    fn from(vec: Vec<[i32; 2]>) -> Self {
-        VertexAttributeValues::Sint32x2(vec)
-    }
-}
-
-impl From<Vec<[u32; 2]>> for VertexAttributeValues {
-    fn from(vec: Vec<[u32; 2]>) -> Self {
-        VertexAttributeValues::Uint32x2(vec)
-    }
-}
-
-impl From<Vec<[f32; 3]>> for VertexAttributeValues {
-    fn from(vec: Vec<[f32; 3]>) -> Self {
-        VertexAttributeValues::Float32x3(vec)
-    }
-}
-
-impl From<Vec<[i32; 3]>> for VertexAttributeValues {
-    fn from(vec: Vec<[i32; 3]>) -> Self {
-        VertexAttributeValues::Sint32x3(vec)
-    }
-}
-
-impl From<Vec<[u32; 3]>> for VertexAttributeValues {
-    fn from(vec: Vec<[u32; 3]>) -> Self {
-        VertexAttributeValues::Uint32x3(vec)
-    }
-}
-
-impl From<Vec<[f32; 4]>> for VertexAttributeValues {
-    fn from(vec: Vec<[f32; 4]>) -> Self {
-        VertexAttributeValues::Float32x4(vec)
-    }
-}
-
-impl From<Vec<[i32; 4]>> for VertexAttributeValues {
-    fn from(vec: Vec<[i32; 4]>) -> Self {
-        VertexAttributeValues::Sint32x4(vec)
-    }
-}
-
-impl From<Vec<[u32; 4]>> for VertexAttributeValues {
-    fn from(vec: Vec<[u32; 4]>) -> Self {
-        VertexAttributeValues::Uint32x4(vec)
-    }
-}
-
-impl From<Vec<[u16; 4]>> for VertexAttributeValues {
-    fn from(vec: Vec<[u16; 4]>) -> Self {
-        VertexAttributeValues::Uint16x4(vec)
-    }
-}
-
-impl From<Vec<[u8; 4]>> for VertexAttributeValues {
-    fn from(vec: Vec<[u8; 4]>) -> Self {
-        VertexAttributeValues::Unorm8x4(vec)
-    }
-}
-
-impl TryFrom<VertexAttributeValues> for Vec<[u8; 4]> {
-    type Error = FromVertexAttributeError;
-
-    fn try_from(value: VertexAttributeValues) -> Result<Self, Self::Error> {
-        match value {
-            VertexAttributeValues::Uint8x4(value) | VertexAttributeValues::Unorm8x4(value) => {
-                Ok(value)
-            }
-            _ => Err(FromVertexAttributeError::new::<Self>(value)),
-        }
-    }
-}
-
-impl TryFrom<VertexAttributeValues> for Vec<[i8; 4]> {
-    type Error = FromVertexAttributeError;
-
-    fn try_from(value: VertexAttributeValues) -> Result<Self, Self::Error> {
-        match value {
-            VertexAttributeValues::Sint8x4(value) | VertexAttributeValues::Snorm8x4(value) => {
-                Ok(value)
-            }
-            _ => Err(FromVertexAttributeError::new::<Self>(value)),
-        }
-    }
-}
-
-impl TryFrom<VertexAttributeValues> for Vec<[u8; 2]> {
-    type Error = FromVertexAttributeError;
-
-    fn try_from(value: VertexAttributeValues) -> Result<Self, Self::Error> {
-        match value {
-            VertexAttributeValues::Uint8x2(value) | VertexAttributeValues::Unorm8x2(value) => {
-                Ok(value)
+macro_rules! impl_from {
+    ($from:ty, $variant:tt) => {
+        impl From<Vec<$from>> for VertexAttributeValues {
```

Comment:
Should we be inlining these trait impls? Do we have benchmarks or stress tests that might be able to spot a difference there?

Reply:
I couldn't find any that update vertex attributes.